File: mobile/android/air/SubModules/UISettings/src/main/java/org/mytonwallet/app_air/uisettings/viewControllers/settings/models/SettingsItem.kt
package org.mytonwallet.app_air.uisettings.viewControllers.settings.models

import org.mytonwallet.app_air.walletcore.models.MAccount

data class SettingsItem(
    val identifier: Identifier,
    val icon: Int? = null,
    val title: String,
    val value: String? = null,
    val hasTintColor: Boolean,
    val account: MAccount? = null
) {
    enum class Identifier {
        ACCOUNT,
        ADD_ACCOUNT,
        APPEARANCE,
        ASSETS_AND_ACTIVITY,
        CONNECTED_APPS,
        LANGUAGE,
        SECURITY,
        WALLET_VERSIONS,
        QUESTION_AND_ANSWERS,
        TERMS,
        SWITCH_TO_LEGACY,
    }

    override fun equals(other: Any?): Boolean {
        if (this == other) return true
        if (other !is SettingsItem) return false
        return identifier == other.identifier
    }

    override fun hashCode(): Int {
        return (identifier.toString() + '_' + account?.accountId).hashCode()
    }
}

File: mobile/ios/Air/SubModules/UISettings/Settings/Models/SettingsItem.swift
//
//  SettingsItem.swift
//  UISettings
//
//  Created by Sina on 6/26/24.
//

import Foundation
import UIKit
import WalletCore
import WalletContext

struct SettingsItem: Equatable, Identifiable {
    
    enum Identifier: Equatable, Hashable {
        case changeAvatar
        case editWalletName
        case account(accountId: String)
        case addAccount
        case appearance
        case assetsAndActivity
        case connectedApps
        case language
        case security
        case backup
        case walletVersions
        case questionAndAnswers
        case terms
        case switchToCapacitor
        case signout
    }
    let id: Identifier
    let icon: UIImage?
    let title: String
    var subtitle: String? = nil
    var value: String? = nil
    let hasPrimaryColor: Bool
    let hasChild: Bool
    let isDangerous: Bool
}
